python/lsst/meas/algorithms/matcherSourceSelector.py

```diff
@@ -0,0 +1,171 @@
+#
+# LSST Data Management System
+# Copyright 2008-2015 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+from __future__ import absolute_import, division, print_function
+
+import numpy as np
+
+from lsst.afw import table
+import lsst.pex.config as pexConfig
+from .sourceSelector import BaseSourceSelectorConfig, BaseSourceSelectorTask, sourceSelectorRegistry
+from lsst.pipe.base import Struct
+from functools import reduce
+
+
+class MatcherSourceSelectorConfig(BaseSourceSelectorConfig):
+    sourceFluxType = pexConfig.Field(
+        doc="Type of source flux; typically one of Ap or Psf",
+        dtype=str,
+        default="Ap",
+    )
+    minSnr = pexConfig.Field(
+        dtype=float,
+        doc="Minimum allowed signal-to-noise ratio for sources used for matching "
+        "(in the flux specified by sourceFluxType); <= 0 for no limit",
+        default=40,
+    )
+
+
+class MatcherSourceSelectorTask(BaseSourceSelectorTask):
+    """
+    !Select sources that are useful for astrometry.
+
+    Good astrometry sources have high signal/noise, are non-blended, and
+    did not have certain "bad" flags set during source extraction. They need not
+    be PSF sources, just have reliable centroids.
+    """
+    ConfigClass = MatcherSourceSelectorConfig
+
+    def __init__(self, *args, **kwargs):
+        BaseSourceSelectorTask.__init__(self, *args, **kwargs)
+
+    def selectSources(self, sourceCat, matches=None):
+        """
+        !Return a catalog of sources: a subset of sourceCat.
+
+        If sourceCat is cotiguous in memory, will use vectorized tests for ~100x
+        execution speed advantage over non-contiguous catalogs. This would be
+        even faster if we didn't have to check footprints for multiple peaks.
+
+        @param[in] sourceCat  catalog of sources that may be sources
+                                (an lsst.afw.table.SourceCatalog)
+
+        @return a pipeBase.Struct containing:
+        - sourceCat  a catalog of sources
+        """
+        self._getSchemaKeys(sourceCat.schema)
+
+        if sourceCat.isContiguous():
+            good = self._isUsable_vector(sourceCat)
```

Comment:
The base class I think already defines all the ones you want in the `badFlags` config parameter. Cosmics are crCenter, I think.

I really do think you should consider what the "correct" source selection criteria are, instead of just reproducing SourceInfo: there's really no guarantee it was fully thought out.